client/src/main/java/org/spine3/validate/Validate.java

```diff
@@ -164,16 +164,22 @@ public static String checkNotEmptyOrBlank(String stringToCheck, String fieldName
     }
 
     /**
-     * Ensures that the passed timestamp is not {@code null} and has {@code seconds} value which is greater than zero.
+     * Ensures that the passed timestamp:
+     * <ul>
+     * <li>is not {@code null};
+     * <li>{@code seconds} > 0;
+     * <li>{@code nanos} >= 0.
+     * </ul>
      *
      * @param timestamp the timestamp to check
+     * @param nameToLog the name of the checked timestamp used in logging
      * @return the passed timestamp
-     * @throws IllegalArgumentException if the timestamp {@code seconds} value is less or equal to zero
+     * @throws IllegalArgumentException if any of the requirements are not met
      */
-    public static Timestamp checkTimestamp(Timestamp timestamp, String fieldName) {
-        checkNotNull(timestamp);
-        final long seconds = timestamp.getSeconds();
-        checkArgument(seconds > 0, fieldName + " is invalid.");
+    public static Timestamp checkIsPositive(Timestamp timestamp, String nameToLog) {
+        checkNotNull(timestamp, nameToLog + " is null.");
+        checkArgument(timestamp.getSeconds() > 0, nameToLog + " must have positive seconds.");
+        checkArgument(timestamp.getNanos() >= 0, nameToLog + " must have positive nanoseconds.");
```

Comment:
`"... must not have negative number of seconds"`.

Please note the condition is NOT strict, i.e. `>=`, so you cannot call it `positive`.